### PR TITLE
added custom apis, all tests passing

### PR DESCRIPTION
**Pull request by aidankmcl**

### CODE REVIEW

File: lib/custom.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  var _this = this;
+  
+  this.collection = collection;
+  if (typeof config !== 'undefined') {
+    config['collection'] = collection
+  }
+  this.baseConfig = config || {'collection': collection};
+  
+  this.addData = function(data, config) {
+    var batch = (typeof data[0] !== 'string') ? true : false;
```
Comment (sihrc) [inline lib/custom.js]: heh. don't really need the true: false here

**Comment by sihrc** [comment]
What's the difference between defining the methods of Collection in the function as attributes as opposed to using `Collection.prototype.method =` ?

**Comment by madisonmay** [comment]
http://stackoverflow.com/questions/3564238/object-oriented-javascript-with-prototypes-vs-closures

File: lib/indico.js [hunk 2]
```diff
@@ -1,127 +1,8 @@
...
-      var size = (api.name === "fer" && config && config["detect"]) ? false : api.size;
-      return image.preprocess(data, size, api.min_axis, batch).then(function (packaged) {
-        return make_request(api.endpoint, packaged, batch, config);
-      });
-    } else {
-      return make_request(api.endpoint, data, batch, config);
-    }
-  };
-}
+var services = require('./services.js').services
```
Comment (madisonmay) [inline lib/indico.js]: Oh yes.  So good.

File: test/integration/custom.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    return;
+  }
+  
+  describe('addData', function() {
+    it('should add a single example to a collection', function(done) {
+      indico.Collection("test").clear().then(function() {
+        var testCollection = indico.Collection("test");
+        
+        testCollection.info().then(function(res) {
+          res['status'].should.equal('no examples');
```
Comment (madisonmay) [inline test/integration/custom.js]: Isn't this promises written in the style of callbacks?  Couldn't we close the first then before we introduce a second.

File: test/integration/custom.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
+var indico = require('../..')
```
Comment (madisonmay) [inline test/integration/custom.js]: Such good tests.

**Comment by madisonmay** [comment]
Node pretty much looks good to me.

**Comment by madisonmay** [comment]
@aidankmcl gonna close and reopen onto development

**Pull request closed without merging**